<?xml version="1.0"?>

<launch>

    <arg name="cmd_vel" default="cmd_vel"/>
    <arg name="joy_dev" default="/dev/input/js_robot"/>
    <arg name="driver" default="ds4drv" />
    <arg name="pad" default="ps4" />

    <arg name="pad_config" default="$(find roboticarts_pad)/config/$(arg driver)_$(arg pad).yaml" />

    <!-- joy node  -->
    <node pkg="joy" type="joy_node" name="joy" output="screen" respawn="true">
        <param name="dev" type="string" value="$(arg joy_dev)" />
    </node>

    <!-- roboticarts_pad file -->
    <node pkg="roboticarts_pad" type="roboticarts_pad_node" name="roboticarts_pad" output="screen">
        <rosparam command="load" file="$(arg pad_config)"       /> 
        <param name="cmd_vel_topic" value="$(arg cmd_vel)"      />
        <param name="pad_settings/driver" value="$(arg driver)" /> 
    </node>

</launch>
